plugins/techdocs/src/reader/components/TechDocsPageHeader.tsx

```diff
@@ -54,6 +54,17 @@ export const TechDocsPageHeader = ({
   const { site_name: siteName, site_description: siteDescription } =
     techdocsMetadataValues || {};
 
+  const docsRootLink = useRouteRef(rootRouteRef)();
+
+  if (!!entityMetadataValues && !!entityMetadataValues.error) {
+    const error = entityMetadataValues.error;
+    if (error.name === 'NotFoundError') {
+      // not found error will be shown in main body
+      return null;
+    }
+    throw error;
+  }
```

Comment:
I think this construct is odd. There's some odd interaction between the page level triggering some loads, and the header drawing or not drawing part of what gets loaded, and the page contents - even before your change - but now it gets even more weird.

The right solution, it seems to me, is to handle the logic outside of the header entirely. The surrounding component should handle the loading and then just inform the header and page what to do, based on the current state of things. For example, if the surrounding page thinks that the header should not be drawn in an error condition, it should just _not render_ the header based on that condition ... instead of the header having to second-guess things and "as a side effect" hiding itself.

Reply:
Agreed, will update

Reply:
Am thinking of changing  to throwing an error inside plugins/techdocs/src/client.ts/getEntityMetadata, as otherwise the return signature for getEntityMetadata will have to be typed to also returning the structure of an error (from request.json() on a request with error). This displays a standard error page.

Reply:
Could be argued that "not found" is different from "there was a critical error in trying to serve the request". We're using the `ErrorPage` in a lot of 404-like circumstances, which by default shows the "Someone dropped the mic" graphic. [Here](https://github.com/backstage/backstage/blob/6494c09b0914255179fcf2872cf9ba11d0bb3558/plugins/scaffolder/src/components/TaskPage/TaskPage.tsx#L290) is an example of that.

So making a hook or API call return `undefined` specifically for not-found cases, but throw a hard error for other cases, makes a lot of sense to me

Reply:
Updated - on 404, getEntityMetadata returns undefined for value and TechDocsPage will render TechDocsNotFound.